streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -406,6 +406,20 @@ void writeAsText(final String filePath,
     void foreach(final ForeachAction<? super K, ? super V> action);
 
     /**
+     * Perform an action on each record of {@code KStream}.
+     * This is a stateless record-by-record operation (cf. {@link #process(ProcessorSupplier, String...)}).
+     *
+     * Peek is a non-terminal operation that triggers a side effect (such as logging or statistics collection)
+     * and returns an unchanged stream.
```

Comment:
It seems to me that the API does not allow this to be violated.  The `ForeachAction` is given only the contents of the stream, and returns nothing.

What kind of test would you propose?